rules/rules.js

```diff
@@ -371,85 +371,104 @@ const SwitchableJSRule = function (ruleConfig) {
   }
 };
 
+/**
+ * Adds a key's value from a Java HashMap to a JavaScript object (as string) if the HashMap has that key.
+ * This function uses the mutable nature of JS objects and does not return anything.
+ * @private
+ * @param {*} hashMap Java HashMap
+ * @param {String} key key from the HashMap to add to the JS object
+ * @param {Object} object JavaScript object
+ */
+const addFromHashMap = (hashMap, key, object) => {
+  if (hashMap.containsKey(key)) object[key] = hashMap[key].toString();
+};
+
+/**
+ * Get rule trigger data from raw Java input and generate JavaScript object.
+ * @private
+ * @param {*} input raw Java input from openHAB core
+ * @returns {rules.EventObject}
+ */
 const getTriggeredData = function (input) {
   const event = input.get('event');
+  const data = {};
 
-  if (event && Java.typeName(event.class) === 'org.openhab.core.items.events.ItemCommandEvent') {
-    return {
-      eventType: 'command',
-      triggerType: 'ItemCommandTrigger',
-      receivedCommand: event.getItemCommand().toString(),
-      oldState: input.get('oldState') + '',
-      newState: input.get('newState') + '',
-      itemName: event.getItemName(),
-      module: input.get('module')
-    };
-  }
+  // Properties of data are dynamically added, depending on their availability
 
-  const ev = event + '';
-  // log.debug("event",ev.split("'").join("").split("Item ").join("").split(" "));
-  let evArr = [];
-  if (ev.includes('triggered')) {
-    const atmp = ev.split(' triggered '); // astro:sun:local:astroDawn#event triggered START
-    evArr = [atmp[0], 'triggered', atmp[1]];
-  } else {
-    evArr = ev.split("'").join('').split('Item ').join('').split(' '); // Item 'benqth681_switch' received command ON
+  // Item triggers
+  if (input.containsKey('command')) {
+    data.receivedCommand = input.get('command').toString();
+    data.command = input.get('command').toString();
+  }
+  addFromHashMap(input, 'oldState', data);
+  addFromHashMap(input, 'newState', data);
+  if (input.containsKey('state')) {
+    data.receivedState = input.get('state').toString();
+    data.state = input.get('state').toString();
```

Comment:
The `state` variable here was from before I did the refactor, I introduced `receivedState` to be more concise with `receivedCommand`.

We can remove `state`, but this would be breaking.

Reply:
I have marked `state` as deprecated and decided to only use `receivedState` which is more concise with `receivedCommand`.

See https://github.com/openhab/openhab-js/pull/136/commits/504c0e90ec329d7877c7c9bfa5f1dc1dc789bf5e.